{<} causes error with orthography rules that remove characters

## Summary
<!-- Briefly, what your is your issue, and how often does it happen (always, sometimes, rarely)? -->
<!-- ↓↓ WRITE SUMMARY BELOW ↓↓ -->
If an stroke isn't explicitly defined in a dictionary but it can be generated (as with `-G`, `-S`, `-Z`, or `-D`), then it will cause an error when it follows `{<}`, but only if the generated result does not use all the characters of the origin entry.

<!-- ↑↑ END SUMMARY ↑↑ -->

## Reproducing
<!-- Usually, in order to help you, the developers need to be able to reproduce issue -->
<!-- What are the steps you take that cause this issue? -->
<!-- You can attach a screenshot of the problem, if it is visual -->
<!-- If Plover is crashing:
          - you can attach the "plover.log" file -->
<!-- If it is a steno output issue:
          - you can turn on stroke logging, reproduce the issue, then attach your "strokes.log" -->
<!-- If it is a machine issue:
          - what machine do you have?
          - how is it connected to your computer? -->
<!-- ↓↓ WRITE REPRODUCE INSTRUCTIONS BELOW ↓↓ -->
Disable all dictionaries. Create one with the following entries:
```
{
"-G": "{^ing}",
"KPA*L": "{<}",
"TKPWAEUPL": "game"
}
```
Write `KPA*L/TKPWAEUPLG`. The expected behaviour is for the string "GAMING" to be produced, but instead the following error shows up:
`ERROR: engine on_stroked failed AssertionError`.
and something like the following shows up in plover.log:
```
2018-06-05 14:05:26,131 [Dummy-1] ERROR: engine on_stroked failed
Traceback (most recent call last):
  File "dist\plover-4.0.0.dev8+8.ge062cda-win32\data\Lib\site-packages\plover\engine.py", line 136, in run
  File "dist\plover-4.0.0.dev8+8.ge062cda-win32\data\Lib\site-packages\plover\engine.py", line 352, in _on_stroked
  File "dist\plover-4.0.0.dev8+8.ge062cda-win32\data\Lib\site-packages\plover\translation.py", line 203, in translate
  File "dist\plover-4.0.0.dev8+8.ge062cda-win32\data\Lib\site-packages\plover\translation.py", line 263, in flush
  File "dist\plover-4.0.0.dev8+8.ge062cda-win32\data\Lib\site-packages\plover\translation.py", line 268, in _output
  File "dist\plover-4.0.0.dev8+8.ge062cda-win32\data\Lib\site-packages\plover\formatting.py", line 367, in format
  File "dist\plover-4.0.0.dev8+8.ge062cda-win32\data\Lib\site-packages\plover\formatting.py", line 475, in render
  File "dist\plover-4.0.0.dev8+8.ge062cda-win32\data\Lib\site-packages\plover\formatting.py", line 425, in render
  File "dist\plover-4.0.0.dev8+8.ge062cda-win32\data\Lib\site-packages\plover\formatting.py", line 401, in _render_action
AssertionError
```

<!-- ↑↑ END REPRODUCE INSTRUCTIONS ↑↑ -->

## Plover Version
<!-- If you can open Plover, view the "About" window and copy the version number -->
<!-- If you are running from source, paste the Git SHA. You can get it by running: `git rev-parse --short HEAD` -->
<!-- ↓↓ WRITE PLOVER VERSION BELOW ↓↓ -->
Plover 4.0.0.dev8+8.ge062cda 

<!-- ↑↑ END PLOVER VERSION ↑↑ -->

## System
<!-- Please tell us if you are running on Windows, Mac, or Linux. -->
<!-- WINDOWS: Are you on Windows 7, 8, 10, etc.? -->
<!-- MAC: Click the Apple in the menu bar, then "About This Mac" to get the version number -->
<!-- LINUX: Which distribution, DE, and WM, and how you are running Plover (deb, AppImage, installed, etc.) -->
<!-- ↓↓ WRITE SYSTEM INFO BELOW ↓↓ -->

Windows 10
<!-- ↑↑ END SYSTEM INFO ↑↑ -->

Blackbox test to reproduce

```python
    def test_carry_upper_bug966(self):
        r'''
        "KPA*L": "{<}",
        "TKPWAEUPL": "game",
        "-G": "{^ing}",

        KPA*L/TKPWAEUPLG ' GAMING'
        '''
```

```
___________ TestsBlackbox.test_carry_upper_bug966 ____________

bb = <test.test_blackbox.TestsBlackbox object at 0x123165eb8>

>   new_f = lambda bb: (f(bb), replay(bb, name, test))

plover_build_utils/testing.py:118:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
plover/translation.py:203: in translate
    self.flush()
plover/translation.py:263: in flush
    self._output(undo, do, prev)
plover/translation.py:268: in _output
    callback(undo, do, prev)
plover/formatting.py:369: in format
    self.spaces_after).render(last_action, old, new)
plover/formatting.py:481: in render
    for action in self.after.render(do, last_action):
plover/formatting.py:427: in render
    self._render_action(action)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <plover.formatting.TextFormatter object at 0x123178470>
action = Action(prev_attach=True, prev_replace='e', text='ING', trailing_space=' ', upper_carry=True, word='gaming')

    def _render_action(self, action):
        if self.spaces_after and self.trailing_space:
            assert self.appended_text.endswith(self.trailing_space)
            self.appended_text = self.appended_text[:-len(self.trailing_space)]
        if action.prev_replace:
            replaced = len(action.prev_replace)
            appended = len(self.appended_text)
            if replaced > appended:
                assert action.prev_replace.endswith(self.appended_text)
                replaced -= appended
                if replaced > len(self.replaced_text):
                    assert action.prev_replace.endswith(self.replaced_text)
                    self.replaced_text = action.prev_replace[:replaced]
                else:
                    assert self.replaced_text.endswith(action.prev_replace)
                    self.replaced_text = self.replaced_text[:-replaced]
                    self.replaced_text += action.prev_replace[:replaced]
                self.appended_text = ''
            else:
>               assert self.appended_text.endswith(action.prev_replace)
E               AssertionError

plover/formatting.py:403: AssertionError
```

--- a/plover/formatting.py
+++ b/plover/formatting.py
@@ -775,7 +775,9 @@ def _apply_meta_attach(meta, ctx):
     ):
         new_word = add_suffix(last_word, meta)
         common_len = len(commonprefix([last_word, new_word]))
-        action.prev_replace = last_word[common_len:]
+        replaced = last_word[common_len:]
+        action.prev_replace = ctx.last_text(len(replaced))
+        assert replaced.lower() == action.prev_replace.lower()
         last_word = last_word[:common_len]
         meta = new_word[common_len:]
     action.text = meta

--- a/test/test_blackbox.py
+++ b/test/test_blackbox.py
@@ -1380,3 +1380,21 @@ class TestsBlackbox(BlackboxTester):
         PHAED/KWREUS  'madies '
         *             'made '
         '''
+
+    def test_bug966_1(self):
+        r'''
+        "-G": "{^ing}",
+        "KPA*L": "{<}",
+        "TKPWAEUPL": "game",
+
+        KPA*L/TKPWAEUPLG  ' GAMING'
+        '''
+
+    def test_bug966_2(self):
+        r'''
+        "-G": "{^ing}",
+        "KPA*L": "{<}",
+        "TKPWAEUPL": "game",
+
+        KPA*L/TKPWAEUPL/-G/*  ' GAME'
+        '''

